lib/matplotlib/cbook.py

```diff
@@ -2487,6 +2490,41 @@ def pts_to_midstep(x, *args):
                    'step-post': pts_to_poststep,
                    'step-mid': pts_to_midstep}
 
+
+def get_index_y(y):
```

Comment:
I don't think either names suit this function, both imply it returns just the index, rather than the indices and the original values...

Which prompts the question, what does this function do that ``enumerate`` does not?